### ISSUE TITLE
bfloat16 cannot be used with tf.keras.backend 

### ISSUE THREAD

**Issue by arahosu** [Knee-Deep-in-MRI/DeepSegmentation#60]
Our current loss functions are implemented using tf.keras.backend. This throws an error when we use more memory-efficient data types like bfloat16 or other mixed-precision strategies. We could potentially reimplement the loss functions using low-level Tensorflow function (without relying on Keras backend functions)
